gateway/ln-gateway/src/lib.rs

```diff
@@ -1182,6 +1185,7 @@ impl Gateway {
                             .await
                             {
                                 warn!("Error registering federation {federation_id}: {e:?}");
+                                Err(GatewayError::FederationError(FederationError::general(e)))?
```

Comment:
Might it make sense to not log here now that we return errors and just let the caller decide at which level to log? That way one of the call-sites wouldn't have to ignore the result entirely.

Reply:
I could see it making sense if there are callers that don't care to log some things, but it could start getting messy if we replaced all logs in this method with errors. For each error they desire to log for, each caller would have to match the error and log

Reply:
+ 1 to Eric's comment. Returning errors would actually make this code a bit more concise. The warnings below could just be returned as errors and the caller can log them